org.eclipse.scanning.points/src/org/eclipse/scanning/points/CompoundGenerator.java

```diff
@@ -33,7 +33,7 @@
  * @author Matthew Gerring
  *
  */
-class CompoundGenerator extends AbstractGenerator<CompoundModel> implements PySerializable {
+class CompoundGenerator extends AbstractGenerator<CompoundModel<Object>> implements PySerializable {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Override the "equals" method in this class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2160)